sdk/identity/identity/src/msal/nodeFlows/msalClient.ts

```diff
@@ -140,29 +144,36 @@ export function createMsalClient(
     pluginConfiguration: generatePluginConfiguration(createMsalClientOptions),
   };
 
-  let confidentialApp: msal.ConfidentialClientApplication | undefined = undefined;
+  const confidentialApps: Map<string, msal.ConfidentialClientApplication> = new Map();
+  // let confidentialApp: msal.ConfidentialClientApplication | undefined = undefined;
   async function getConfidentialApp(
-    _options: GetTokenOptions = {},
+    options: GetTokenOptions = {},
   ): Promise<msal.ConfidentialClientApplication> {
-    // abort requests
-
-    if (confidentialApp === undefined) {
-      // TODOs:
-      // CAE / non-CAE
-      confidentialApp = new msal.ConfidentialClientApplication({
-        ...state.msalConfig,
-        broker: { nativeBrokerPlugin: state.pluginConfiguration.broker.nativeBrokerPlugin },
-        cache: { cachePlugin: await state.pluginConfiguration.cache.cachePlugin },
-      });
-    }
+    const appKey = options.enableCae ? "CAE" : "default";
+
+    if (confidentialApps.has(appKey)) return confidentialApps.get(appKey)!;
```

Comment:
the other way to write this to avoid the `!` is

```suggestion
    const existingApp =  confidentialApps.get(appKey);
    if (existingApp) {
       return existingApp;
     }
```

it's not as compact though.

Reply:
I can change it! It'll probably be more readable anyway than what I have here. Thanks!